flink-clients/src/main/java/org/apache/flink/client/deployment/executors/AbstractJobClusterExecutor.java

```diff
@@ -67,6 +68,10 @@ public AbstractJobClusterExecutor(@Nonnull final ClientFactory clusterClientFact
             @Nonnull final Configuration configuration,
             @Nonnull final ClassLoader userCodeClassloader)
             throws Exception {
+
+        LOG.warn(
+                "Job Clusters are deprecated since Flink 1.15. Please use an Application Cluster/Application Mode instead.");
```

Comment:
I think we now have 2 log statements in the client, and none on the server.
This class is used on the client , and you later call deployJobCluster, which logs another warning.

To cover the server you need to modify the entrypoint accordingly.

Reply:
Good catch. My plan was actually Descriptor and Entrypoint. This is a copy-paste-error, will fix.